batch/batch/aioclient.py

```diff
@@ -0,0 +1,328 @@
+import math
+import time
+import random
+import yaml
+import cerberus
+import aiohttp
+
+from . import schemas
```

Comment:
What is your reasoning for making this a separate file rather than modifying `client.py`?

Reply:
I didn't want to force async on all clients.  For example, I don't think we want to make the pipeline Python api async.  Most users don't want to work with that.

Reply:
I think what I would have done would be to have the implementation be asynchronous and provide a second BatchClient interface that ran everything synchronously to avoid code duplication. Something like this:

```
class BatchClient:
    def __init__(self, ...):
        pass

    async def create_job(self, parameters):
        await self._create_job(parameters)

def run_synchronous(f, args,kwargs): # make this a decorator?
    loop = asyncio.get_event_loop()
    loop.run_until_complete(f, args, kwargs)

class BatchClientSynchronous:
    def __init__(self, ....):
         self._bc = BatchClient()

     def create_job(self, parameters):
          return run_synchronous(self._bc.create_job, parameters)
```

I'll review what you have for now though.

Reply:
Hmm.  In spite of the code duplication, I have to say, I really don't want the core of the synchronous library to be asynchronous.

Reply:
I vote for merging this in as is, and thinking more about how/if we should eliminate the [a]sync duplication.

Reply:
That's fine with me as long as @danking @akotlar are aware when modifying the batch client, that changes may need to be made in both places!